include/utils/color.hpp

```diff
@@ -162,4 +166,89 @@ struct rgba {
   }
 };
 
+// Represent HSL color space colors, with h values in the set [0, 360] and s, l values in the set [0, 1]
+struct hsl {
+  double h;
+  double s;
+  double l;
+  
+  explicit hsl(double h, double s, double l) : h(h), s(s), l(l) {}
+
+  void get_rgb(double& r, double& g, double& b) {
+    if (s == 0.0) {
+      r = g = b = l;
+    } else {
+      const auto hue2rgb = [&](double p, double q, double t) {
+        if (t < 0.0) t += 360.0;
+        if (t > 360.0) t -= 360.0;
+        if (t < 60.0) return p + (q - p) * t/60.0;
+        if (t < 180.0) return q;
+        if (t < 240.0) return p + (q - p) * (240.0 - t)/60.0;
+        return p;
+      };
+
+      auto q = l < 0.5 ? l * (1.0 + s) : l + s - l * s;
+      auto p = 2.0 * l - q;
+      r = hue2rgb(p, q, h + 120.0);
+      g = hue2rgb(p, q, h);
+      b = hue2rgb(p, q, h - 120.0);
+    }
+  }
+
+  rgb to_rgb() {
+    double r, g, b;
+    get_rgb(r, g, b);
+
+    return rgb(r, g, b);
+  }
+
+  rgba to_rgba(double a) {
+    double r, g, b;
+    get_rgb(r, g, b);
+    return rgba(r, g, b, a);
+  }
+
+  static hsl from_rgb(rgb color) {
```

Comment:
I think we are never going to need this direction of the conversion. Because once hsl(a) values are parsed from the config, they should be in the argb format and should never be converted back to hsl.